### DISCUSSION TITLE
Stackoverflow markdown render error

### DISCUSSION THREAD

**Discussion by jayvdb** [lowlighter/metrics#590] [Q&A] [answered]
For 4 of 5 stackoverflow answers on https://github.com/jayvdb/jayvdb, the answer text is

> An error occured:code.replace is not a function
> Please report this to https://github.com/markedjs/marked

I havent reported it there (yet) as I first wanted to check if others are seeing this, as 4 out of 5 is quite high.  Possibly/probably the stackoverflow markup is a particular flavour that https://github.com/markedjs/marked may not support.  Maybe another library is more appropriate?

Also if it is happening so frequently, maybe this error should be detected and the error hidden.

**Comment by lowlighter** [comment]
I'm not aware of other cases (or else it was not reported in discussion/issues)

Stackoverflow has a special syntax for runnable code-blocks IIRC, so it may be related.
Since the message explicitely seems to come from marked library, I'd say open an issue there maybe with a repro if you have some time.

As marked is still supported by its maintainers I'd prefer sticking with it for now (if we open an issue there and it doesn't get fix we could start consider changing it)

> Also if it is happening so frequently, maybe this error should be detected and the error hidden.

Not sure if it's easily doable though, it doesn't seem to explicitely thrown an `Error` but instead returning text with an error message in it.
So to detect it we could possibly use a RegExp, but seems a bit hacky

**Comment by jayvdb** [comment, marked as answer]
It looks like https://github.com/lowlighter/metrics/commit/6126a8e9f29045b026edd03eb9078a08af4dcba0 has fixed it.
